### ISSUE TITLE
onMentionSelected hook

### ISSUE THREAD

**Issue by mxstbr** [withspectrum/slate-mentions#2]
This would allow developers to specify how to render their selected mention. It should also be possible to allow them to make the `suggestions` array data any data structure they want, since they can just format it at the end to fit into the editor.

**Comment by oyeanuj** [comment]
I think a corresponding function that you might need is `onMentionBackspace`. The default can be the function that you have right now but the user can override it.

**Comment by mxstbr** [comment]
Agree, what about calling them `onMentionSelection` and `onMentionDeletion`?

**Comment by oyeanuj** [comment]
Maybe `onMentionInsertion` instead of `selection` (since arrow keys are technically selecting it?)
